pkg/client/openstack/compute/create.go

```diff
@@ -0,0 +1,43 @@
+package compute
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack/compute/v2/servers"
+)
+
+type createError struct {
+}
+
+func (ce *createError) Error400(e gophercloud.ErrUnexpectedResponseCode) error {
+
+	var response struct {
+		BadRequest struct {
+			Message string `json:"message"`
+			Code    int    `json:"code"`
+		} `json:"badRequest"`
+	}
+	if err := json.Unmarshal(e.Body, &response); err != nil || response.BadRequest.Message == "" {
+		return fmt.Errorf("Failed to parse response: %s", string(e.Body))
+	}
+	return fmt.Errorf("Response from compute: %d %s", e.Actual, response.BadRequest.Message)
+}
+
+func (ce createError) Error() string {
+	return "Failed to create server. This should not never be printed."
```

Comment:
not never == always?

Reply:
The deserialisation has to be defined for each error code. If quota errors result in a 400 bad request its already covered. If quota errors return something different we need to implement an additional interface for it.

Reply:
Ups. You got me. this doesn't make sense. not negation to much.